.github/workflows/build.yml

```diff
@@ -0,0 +1,73 @@
+name: build
+
+on:
+  push:
+    branches: [ master, '[0-9]+.[0-9]+', '[0-9]+.[0-9]+.[0-9]+' ]
+  pull_request:
+    branches: [ master, '[0-9]+.[0-9]', '[0-9]+.[0-9]+.[0-9]+' ]
+
+jobs:
+  build:
+    if: "!contains(github.event.head_commit.message, '[skip ci]')"
+    name: "PHP ${{ matrix.php-versions }} Test (deps: ${{ matrix.dependencies }})"
+
+    continue-on-error: ${{ matrix.experimental }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        php-versions: ['7.2', '7.3', '7.4']
+        dependencies: ['highest']
+        composer-arguments: [''] # to run --ignore-platform-reqs in experimental builds
+        experimental: [false]
+        include:
+          # Experimental build for PHP 8
+          - php-versions: '8.0'
+            composer-arguments: '--ignore-platform-reqs'
+            experimental: true
+            dependencies: 'highest'
+
+          # Build for minimum dependencies. Note: Running tests with PHP 5.5 fails due to PHPUnit PHP requirements
+          - php-versions: '7.2'
+            experimental: false
+            dependencies: 'lowest'
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php-versions }}
+          extensions: mbstring, json
+          coverage: xdebug #optional
+          ini-values: date.timezone="Africa/Tunis", opcache.fast_shutdown=0
+
+      - name: Composer install
+        uses: ramsey/composer-install@v1
+        with:
+          dependency-versions: ${{ matrix.dependencies }}
+          composer-options: ${{ matrix.composer-arguments }}
+
+      - name: Run tests
+        run: vendor/bin/phpunit --coverage-clover=build/logs/clover.xml
+
+      - name: Coveralls
+        run: vendor/bin/php-coveralls -v
+        env:
+          COVERALLS_RUN_LOCALLY: 1
+          COVERALLS_REPO_TOKEN: ${{ secrets.COVERALLS_REPO_TOKEN }}
+
+  # Use this step to configure a required build pass via gh branch protection rules (currently makes experimental builds unusable due to https://github.com/actions/toolkit/issues/581)
```

Comment:
@NamelessCoder if you want to require build passes for the matrix independent of the specific configuration, you can outcomment this section and then add a branch protection rule for this build. However, due to the mentioned issue, this would make the experimental failing steps fail this buildall, so that would need to be disabled (we did this in Neos, because the supported PHP versions differ between branches, so we'd need to have specific branch protection rules for every single branch instead of our pattern)